ecl/hthor/hthorkey.cpp

```diff
@@ -2597,7 +2598,7 @@ class XmlFetchPartHandler : public SimpleFetchPartHandlerBase, public IXMLSelect
                 part->getFilename(rfn).getPath(fname);
                 throw owner.makeWrappedException(e, fname.str());
             }
-            if(!gotNext)
+            if(!gotNext && !lastMatch) //unfortunately json parser next() has slightly different behavior, tricky, may fix later
```

Comment:
This comment doesn't explain how it is different or why the condition needs to check for !lastMatch.  You may not remember if you ever revisit it.
Is this something that is likely to be worth changing?